proptest/src/num/float_samplers.rs

```diff
@@ -0,0 +1,338 @@
+//-
+// Copyright 2022 Michael Wright
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Alternative uniform float samplers because the ones provided by the rand crate are prone
+//! to overflow. The samplers work by uniformly selecting from a set of equally spaced values in
+//! the interval and the included bounds. Selection is slighly biased towards the bounds.
+
+pub(crate) use self::f32::F32U;
+pub(crate) use self::f64::F64U;
+
+macro_rules! float_sampler {
+    ($typ: ident, $int_typ: ident, $wrapper: ident) => {
+        pub mod $typ {
+            use rand::prelude::*;
+            use rand::distributions::uniform::{
+                SampleBorrow, SampleUniform, Uniform, UniformSampler,
+            };
+
+            #[must_use]
+            // Returns the previous float value. In other words the greatest value representable
+            // as a float such that `last_float(a) < a`. `-0.` is treated as `0.`.
+            fn last_float(a: $typ) -> $typ {
+                debug_assert!(a.is_finite() && a > $typ::MIN, "`last_float` invalid input: {}", a);
+                if a == (0.) {
+                    -$typ::from_bits(1)
+                } else if a < 0. {
+                    $typ::from_bits(a.to_bits() + 1)
+                } else {
+                    $typ::from_bits(a.to_bits() - 1)
+                }
+            }
+
+            #[must_use]
+            fn last_gap(a: $typ) -> $typ {
```

Comment:
i suggest `clamped_float_remainder` or something along those lines.

Reply:
given my other comment update, maybe `magnitude_remainder` as a name?

Reply:
I do like the term `gap` though perhaps `difference` is more precise? I'm not a huge fan of `magnitude_remainder` on its own.

following on my preferences from other comment, i'd say something like `difference_between_minimally_smaller_float` or `difference_to_minimally_smaller_float`?

Reply:
`difference_to_minimally_smaller_float` sounds good to me 👍

Reply:
Interesting! I'm in favor of using that -- wasn't aware there was an agreed upon name for this already. Thanks for sharing

Reply:
Wasn't aware of that either, 👍